src/celldega/pre/image_quantification.py

```diff
@@ -0,0 +1,558 @@
+                        pixel_values = pixel_values[pixel_values != src.nodata]
+
+                        mean_intensity = pixel_values.mean() if len(pixel_values) > 0 else 0
+                        sum_intensity = np.sum(pixel_values) if len(pixel_values) > 0 else 0
+                    except ValueError:
+                        mean_intensity, sum_intensity = 0, 0
+
+                    intensities.append(mean_intensity)
+                    sum_intensities.append(sum_intensity)
+
+    elif technology_name == 'MERSCOPE':
+
+        with rasterio.open(image_array) as src:
+            image_crs = src.crs
+
+            if cell_polygons.crs != image_crs:
+                cell_polygons = cell_polygons.to_crs(image_crs)
+
+            intensities, sum_intensities = [], []
+            for polygon in cell_polygons.geometry:
+                try:
+                    out_image, _ = mask(src, [polygon], crop=True)
+                    pixel_values = out_image[0].flatten()
+                    pixel_values = pixel_values[pixel_values != src.nodata]
+
+                    mean_intensity = pixel_values.mean() if len(pixel_values) > 0 else 0
+                    sum_intensity = np.sum(pixel_values) if len(pixel_values) > 0 else 0
+                except ValueError:
+                    mean_intensity, sum_intensity = 0, 0
+
+                intensities.append(mean_intensity)
+                sum_intensities.append(sum_intensity)
+
+    cell_polygons["mean_stain_intensity"] = intensities
+    cell_polygons["sum_stain_intensity"] = sum_intensities
+
+    return cell_polygons, image_array
+
+def plot_zonal_stats(image_array, cell_polygons):
+
+    """
+    Plots spatial and statistical distributions of stain intensities across cell polygons.
+
+    Parameters:
+    -----------
+    image_array : numpy.ndarray
+        A 2D or 3D array representing the stain image on which the cell polygons will be overlaid.
+    cell_polygons : gpd.GeoDataFrame
+        A GeoDataFrame containing cell polygon geometries and their corresponding "sum_stain_intensity" values.
+
+    Returns:
+    --------
+    tuple:
+        - numpy.ndarray: Log-transformed sum stain intensities for each cell polygon.
+        - numpy.ndarray: Bin edges of the histogram for log-normalized stain intensities.
+
+    Plots:
+    ------
+    1. An overlay of cell polygons with stain-positive intensities on the stain image.
+    2. A histogram of log-normalized stain intensities, showing their distribution.
+
+    """
+
+    log_transformed_stain_intensities = np.log1p(cell_polygons['sum_stain_intensity'])
+
+    # Normalize the log-transformed intensities to the range [0, 1]
+    log_normalized_stain_intensities = (log_transformed_stain_intensities - log_transformed_stain_intensities.min()) / \
+                                     (log_transformed_stain_intensities.max() - log_transformed_stain_intensities.min())
+
+    print("Plotting cells with total stain intensity greater than zero:")
```

Comment:
If I understand correctly, `total `stain` here means the cumulative pixel number (or brightness values) within a cell, which seems depending on the size of a cell, while `mean` or `median` value of the intensities is more like a `average` representation of the expression at cell level. What is your thought?

Thanks!